sky/exceptions.py

```diff
@@ -120,7 +120,35 @@ def deserialize_exception(serialized: Dict[str, Any]) -> Exception:
     if exception_class is None:
         # Unknown exception type.
         return Exception(f'{exception_type}: {serialized["message"]}')
-    e = exception_class(*serialized['args'], **serialized['attributes'])
+
+    # Try to construct the exception with both args and attributes.
+    # Built-in exceptions (like RuntimeError, ValueError) don't accept
+    # keyword arguments, so we fall back to constructing with args only
+    # and then setting attributes manually.
+    try:
+        e = exception_class(*serialized['args'], **serialized['attributes'])
+    except TypeError:
+        # Built-in exceptions don't accept keyword arguments.
+        # Construct with args only and set attributes manually.
+        try:
+            e = exception_class(*serialized['args'])
+        except Exception as deserialize_err:  # pylint: disable=broad-except
+            # If construction with positional args also fails, the serialized
+            # data is likely incompatible. Fall back to a generic Exception
+            # to prevent crashing the deserialization process, which could
+            # mask the original error.
+            return Exception(
```

Comment:
Let's use a dedicated exception for this, e.g. `SkyPilotInternalDeserializationError`, to make it easier to track